datastore/key.go

```diff
@@ -105,19 +105,29 @@ func (k *Key) marshal(b *bytes.Buffer) {
 	} else {
 		b.WriteString(strconv.FormatInt(k.ID, 10))
 	}
-	if k.Namespace != "" {
+	if k.Namespace != "" && includeNamespace {
 		b.WriteByte(',')
 		b.WriteString(k.Namespace)
 	}
 }
 
-// String returns a string representation of the key.
+// String returns a string representation of the key. It does not include
+// the namespace in case that the Namespace's name is sensitive information.
 func (k *Key) String() string {
+	return k._string(false)
+}
+
+// StringWithNamespace is identical to String(), but appends the namespace.
+func (k *Key) StringWithNamespace() string {
```

Comment:
Better name for StringWithNamespace.
May be StringifyInternal or StringInternal or something else

Reply:
If it's truly meant to be "internal", then should we just make it non-exported and call it `stringInternal`?